server/remote/msgraph/get_default_calendar_view.go

```diff
@@ -30,16 +30,7 @@ type calendarViewBatchResponse struct {
 }
 
 func (c *client) GetDefaultCalendarView(remoteUserID string, start, end time.Time) ([]*remote.Event, error) {
-	paramStr := getQueryParamStringForCalendarView(start, end)
-
-	res := &calendarViewResponse{}
-	err := c.rbuilder.Users().ID(remoteUserID).CalendarView().Request().JSONRequest(
-		c.ctx, http.MethodGet, paramStr, nil, res)
-	if err != nil {
-		return nil, errors.Wrap(err, "msgraph GetDefaultCalendarView")
-	}
-
-	return res.Value, nil
+	return c.GetEventsBetweenDates(remoteUserID, start, end)
```

Comment:
It seems like these two methods are just copy of each other. Maybe we should just use `GetDefaultCalendarView`? Or keep `GetEventsBetweenDates` as it's more descriptive. Do we need two different implementations for gcal?

Reply:
In my head, we should have a `GetEvents` that accepts filtering (right now it is this `GetEventsBetweenDates`), and then having a `views` file with custom methods for specific defined views (like the calendar, today, tomorrow ones) or either move that methods to the main "core" implementation. But I'm leaving that as a nice-to-have, since it doesn't specifically targets something related to the current task.

Reply:
I suppose what I'm concerned about is that these are generally doing the same thing right now, though the implementation is slightly different (meaning not direct copy & paste, though seems to do the same high-level operation). Any reason not to have them share functionality by having this method call the existing one? Mainly trying to avoid duplicated code getting out of sync. I trust the direction you have here.

Reply:
Assuming you mean "this method = `GetEventsBetweenDates`", since `GetDefaultCalendarView` is just a view with a predefined filter/paremeters, I think this is the one that should call the sightly lower-level method that actually retrieves the data, since the "view" only specifies parameters in my head, the same way a `GetTodayView`, `GetTomorrowView` would for other commands.